### PR TITLE
Daemonize celery

### PR DESCRIPTION
**Pull request by fruttasecca**
# Description
This PR makes it so that the celery worker(s) process will be respawned if, for any reason, stops. This is done by launching celery through `supervisor`.

### Checklist
- [X] I have manually tested the application to make sure the changes don’t cause any downstream issues, which includes making sure `./orchest status --ext` is not reporting failures when Orchest is running.

### CODE REVIEW

**Review by ricklamers — APPROVED** [review]

File: services/orchest-api/Dockerfile_celery [hunk 1]
```diff
@@ -2,20 +2,32 @@ FROM python:3.8-slim
...
+COPY ./requirements.txt /orchest/services/orchest-api/
 COPY ./lib/python /orchest/lib/python
 
 # Set the `WORKDIR` so the editable installs in the `requirements.txt`
 # can use relative paths.
 WORKDIR /orchest/services/orchest-api/
 RUN pip3 install -r requirements.txt
 
+# Needed to daemonize celery workers.
+RUN apt-get update && apt-get install -y supervisor && mkdir -p /var/log/supervisor
```
Comment (yannickperrenet) [inline services/orchest-api/Dockerfile_celery:14]: Can we move this `RUN` command to the top of the Dockerfile? To optimize for build speed.

File: services/orchest-api/Dockerfile_celery [hunk 2]
```diff
@@ -2,20 +2,32 @@ FROM python:3.8-slim
...
 WORKDIR /orchest/services/orchest-api/app/
+
 ARG ORCHEST_VERSION
 ENV ORCHEST_VERSION=${ORCHEST_VERSION}
-CMD ["./../start_celery_workers.sh"]
+ENTRYPOINT ["/usr/bin/supervisord"]
+# -n to run in the foreground, -m is the umask of the supervisord
+# process, the umask needs to be setup for each child/service process.
+# -u 0 is to run as root.
+CMD ["-n", "-m 002", "-u 0"]
```
Comment (yannickperrenet) [inline services/orchest-api/Dockerfile_celery:33]: I think we should add this `CMD` to the `ENTRYPOINT`, if we later choose to add a CMD when starting the `celery-worker` we don't want to be overwriting this `CMD`.

File: services/orchest-api/celery_daemon_configs/worker_builds.conf [hunk 3]
```diff
@@ -0,0 +1,19 @@
+[program:worker_builds]
+user=root
```
Comment (yannickperrenet) [inline services/orchest-api/celery_daemon_configs/worker_builds.conf:2]: Why is it run as root? Not that I am against it, just for my information.

**Comment by yannickperrenet** [comment]
I like this very much! :racehorse:

**Review by yannickperrenet — APPROVED** [review]

**Pull request merged**